extensions/amp-next-page/1.0/service.js

```diff
@@ -213,19 +213,16 @@ export class NextPageService {
    */
   updateVisibility() {
     this.pages_.forEach((page, index) => {
-      if (
-        page.relativePos === ViewportRelativePos.INSIDE_VIEWPORT ||
-        page.relativePos === ViewportRelativePos.CONTAINS_VIEWPORT
-      ) {
+      if (page.relativePos === ViewportRelativePos.OUTSIDE_VIEWPORT) {
+        if (page.isVisible()) {
+          page.setVisibility(VisibilityState.HIDDEN);
+        }
+      } else {
```

Comment:
The original code did nothing for `ViewportRelativePos.LEAVING_VIEWPORT`, is it intended now for that state to be accounted for in the `else` statement now?

Reply:
Yup this intentionally counts partially visible pages as visible so that images and other elements start rendering as soon as the page enters the view (as opposed to the current behavior where they only render after the document goes past the top of the page)